test/connectors/AuthConnectorSpec.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import com.github.tomakehurst.wiremock.client.WireMock._
+import config.WSHttp
+import connectors.AuthConnector.InvalidCredentials
+import model.{BearerToken, LoginDetails, Role}
+import org.joda.time.DateTime
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{BeforeAndAfterEach, Matchers}
+import play.api.libs.json.Json
+import uk.gov.hmrc.crypto.Protected
+import uk.gov.hmrc.play.http.HeaderCarrier
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+
+
+class AuthConnectorSpec extends UnitSpec with Matchers with ScalaFutures with WiremockSugar with BeforeAndAfterEach with WithFakeApplication {
+
+  override def beforeEach() {
+    startMockServer()
+  }
+
+  override def afterEach() {
+    stopMockServer()
+  }
```

Comment:
I literally had no idea this was valid scala syntax.  I guess it wouldn't be a kitchen sink language without it ;)